### PR TITLE
Override generated scenarios

### LINKED ISSUE #30: Override specific scenarios via template?
I need to use backstop-crawl to regenerate my test senarios from time to time.

This is my backstop.template.json
```
{
  "viewports": [
    {
      "label": "phone",
      "width": 320,
      "height": 480
    },
    {
      "label": "tablet",
      "width": 1024,
      "height": 768
    },
    {
      "name": "desktop",
      "width": 1440,
      "height": 900
    }
  ],
  "defaultScenario": {
    "label": "",
    "url": "",
    "referenceUrl": "",
    "selectors": ["document"],
    "hideSelectors": [],
    "delay": 1500,
    "misMatchThreshold": 0.1
  },
  "scenarios": [],
  "paths": {
    "bitmaps_reference": "backstop_data/bitmaps_reference",
    "bitmaps_test": "backstop_data/bitmaps_test",
    "casper_scripts": "backstop_data/casper_scripts",
    "html_report": "backstop_data/html_report",
    "ci_report": "backstop_data/ci_report"
  },
  "casperFlags": [],
  "engine": "puppeteer",
  "report": ["browser"],
  "debug": false
}
```

I know that a handful of my scenarios fail unless I up the delay to `3000` instead of `1500`.

Is it possible to modify this template in some way so that when I run backstop crawl and generate my backstop.js, those cases use a different delay value than the default?

### PR DESCRIPTION
**Pull request by ghost**
This PR probably needs a little work. I'm new to contributing to other people's projects, so criticism is welcome!

First of all, I had to update `nyc` and `ava` to get the tests to run. Then I noticed two of the tests were failing because I'm running the project on my Windows machine. I updated the tests so that they are now Windows-compatible.

Normally, backstop-crawl allows the user to define a `backstop.template.json`, however, I wanted to ability to explicitly state which template file to use.

I added the option to pass in a `--template` flag. If the path is invalid, an error will be shown. If the flag isn't provided, backstop-crawl behaves as usual.

I also added functionality to process scenario overrides. If the user defines scenarios in their template, those will override generated scenarios with the same label.

If there are no matching labels, the override is ignored.

An idea I had that might be nice is to have overridden scenarios AND additional scenarios. I figured this could be tackled later.

This ticket resolves #30

### CODE REVIEW

File: lib/process-overrides.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+module.exports = processOverrides;
+function processOverrides(urls, scenarios) {
+
+    scenarios.forEach((scenario, scenarioKey) => {
```
Comment (ghost) [inline lib/process-overrides.js:4]: I know that doing a forEach inside another forEach isn't ideal, but I haven't yet figured out a more efficient way to do this.

File: README.md [hunk 2]
```diff
@@ -62,6 +64,10 @@ For example:
 }
 ```
 
+After running generating your `backstop.json`, you can modify scenarios and copy them into `scenarios` in your template. The next time you run `backstop crawl`, these scenarios will override generated scenarios with a matching label.
```
Comment (TI-dejesus) [inline README.md]: Running or generating?

File: lib/crawl.js [hunk 3]
```diff
@@ -80,6 +74,24 @@ function crawl(url, flags) {
...
+
+    /**
+     * Stash the default scenario from the template then delete it
+     * as defaultScenario isn't a valid key for backstop.json
+     */
+    const defaultScenario = defaultConf.defaultScenario;
+    delete defaultConf.defaultScenario;
+
+    // Grab any scenarios defined in the template.
+    overriddenScenarios = defaultConf.scenarios;
```
Comment (TI-dejesus) [inline lib/crawl.js:93]: Just to make sure, `scenarios` is part of the default template right?

File: lib/crawl.js [hunk 4]
```diff
@@ -138,6 +150,9 @@ function crawl(url, flags) {
             });
             urls = limitSimilar(urls, flags.limitSimilar);
         }
+
+        urls = processOverrides(urls, overriddenScenarios);
```
Comment (TI-dejesus) [inline lib/crawl.js:154]: This should always be processed? Or should this be processed if `flag.template` exists?

Comment (ghost) [reply lib/crawl.js:154]: Should always. They might have overriden scenarios defined in `backstop.template.json`.

File: lib/crawl.js (hunk 3 above)
Comment (ghost) [reply lib/crawl.js:93]: Correct. It's an empty array by default.

File: README.md (hunk 2 above)
Comment (ghost) [reply README.md]: Whoops. Fixed.

**Pull request closed without merging**